File: CS50 Problem Sets/Pset1/Mario_more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
	// a program to print a double pyramid to a specified height
	
    int height;
    
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);
    
    for (int i = 0; i < height; i++)
    {
        // loop for spaces
        for (int spaces = height - 1; spaces > i; spaces--)
        {
            printf(" ");
        }
        
        // loop for first set of hashes
        for (int hashes = 1; hashes < i + 1; hashes++)
        {
            printf("#");
        }
        
      printf("#  #");
        
        // loop for second set of hashes
        for (int hashes = 1; hashes < i + 1; hashes++)
        {
            printf("#");
        }
        
      printf("\n");
        
    }
}

File: Pset6/mario.py
# python has no do while loop


def main():
    # invoke a infinite loop and break when condition is met
    while True:
        height = int(input("Height: "))
        if height >= 1 and height <= 8:
            break

    for i in range(height):

        # loop for spaces
        spaces = height - 1
        for j in range(spaces - i):
            print(" ", end="")

        # loop for first set of hashes
        for m in range(i + 1):
            print("#", end='')

        print("  ", end='')

        # loop for second set of hashes
        for k in range(i + 1):
            print("#", end="")
        print()


if __name__ == "__main__":
    main()
